src/main/java/http_protocol/MIMETypes.java

```diff
@@ -0,0 +1,18 @@
+package http_protocol;
+
+import java.util.HashMap;
+
+public class MIMETypes {
+    private static HashMap<String, String> types = new HashMap<>();
+
+    public static String getFileType(String mimeType) {
+        types.put("text/plain", "txt");
```

Comment:
Seems a bit odd that this function puts everything in the hash map and gets something out of it. Then every time you call `getFileType`, it's trying to add to the hash map before retrieving from the hash map.